tests/v2/validation/provisioning/rke1/etcd_node_delete_test.go

```diff
@@ -0,0 +1,124 @@
+)
+
+type EtcdNodeDeleteTestSuite struct {
+	suite.Suite
+	session                *session.Session
+	client                 *rancher.Client
+	ns                     string
+	rke1kubernetesVersions []string
+	cnis                   []string
+	providers              []string
+	nodesAndRoles          []nodepools.NodeRoles
+}
+
+func (e *EtcdNodeDeleteTestSuite) TearDownSuite() {
+	e.session.Cleanup()
+}
+
+func (e *EtcdNodeDeleteTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	e.session = testSession
+
+	e.ns = defaultNS
+
+	clustersConfig := new(provisioning.Config)
+	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
+
+	e.rke1kubernetesVersions = clustersConfig.RKE1KubernetesVersions
+
+	e.cnis = clustersConfig.CNIs
+	e.providers = clustersConfig.Providers
+	e.nodesAndRoles = clustersConfig.NodesAndRolesRKE1
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(e.T(), err)
+
+	e.client = client
+}
+
+func (e *EtcdNodeDeleteTestSuite) TestEtcdDeleteNode() {
+	require.GreaterOrEqual(e.T(), len(e.providers), 1)
+	require.GreaterOrEqual(e.T(), len(e.cnis), 1)
+	require.GreaterOrEqual(e.T(), len(e.rke1kubernetesVersions), 1)
+
+	subSession := e.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := e.client.WithSession(subSession)
+	require.NoError(e.T(), err)
+
+	for _, provider := range e.providers {
+		for _, k8sVersion := range e.rke1kubernetesVersions {
+			for _, cni := range e.cnis {
+				provider := CreateProvider(provider)
+				nodeTemplate, err := provider.NodeTemplateFunc(client)
+				require.NoError(e.T(), err)
+
+				clusterResp, err := TestProvisioningRKE1Cluster(e.T(), client, provider, e.nodesAndRoles, k8sVersion, cni, nodeTemplate)
+				require.NoError(e.T(), err)
+
+				machines, err := e.client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
+					"clusterId": clusterResp.ID,
+				}})
+				require.NoError(e.T(), err)
+
+				numOfEtcdNodesBeforeDeletion := 0
+				etcdNodeToDelete := management.Node{}
+
+				for _, machine := range machines.Data {
+					if machine.Etcd {
+						etcdNodeToDelete = machine
+						numOfEtcdNodesBeforeDeletion++
+					}
+				}
+
+				logrus.Info("Deleting, " + etcdNodeToDelete.NodeName + " etcd node..")
+				err = e.client.Management.Node.Delete(&etcdNodeToDelete)
+				require.NoError(e.T(), err)
+
+				err = clusters.WaitClusterToBeUpgraded(client, clusterResp.ID)
+				require.NoError(e.T(), err)
+
+				err = nodestat.IsNodeReady(client, clusterResp.ID)
+				require.NoError(e.T(), err)
+
+				isEtcdNodeReplaced, err := nodestat.IsRKE1EtcdNodeReplaced(client, etcdNodeToDelete, clusterResp, numOfEtcdNodesBeforeDeletion)
+				require.NoError(e.T(), err)
+
+				require.True(e.T(), isEtcdNodeReplaced)
+
+				podResults, podErrors := pods.StatusPods(client, clusterResp.ID)
+				assert.NotEmpty(e.T(), podResults)
+				assert.Empty(e.T(), podErrors)
+
+			}
+		}
+	}
+}
+
+func TestEtcdDeleteNode(t *testing.T) {
```

Comment:
I am still a little confused on the test expectations based on the naming of these files / test suite. I understand that we are deleting the node, but since we also test that it is replaced, we should update all names to reflect that.

Reply:
yeah that sounds great

Reply:
Made the required change, renamed files and test suites to TestEtcdDeleteAndReplace.